x-pack/plugins/enterprise_search/public/applications/app_search/components/engine_overview_header/engine_overview_header.test.tsx

```diff
@@ -0,0 +1,75 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License;
+ * you may not use this file except in compliance with the Elastic License.
+ */
+
+import React from 'react';
+import { mount } from 'enzyme';
+
+import { EngineOverviewHeader } from '../engine_overview_header';
+import { KibanaContext } from '../../..';
+
+describe('EngineOverviewHeader', () => {
+  let enterpriseSearchUrl;
+
+  afterEach(() => {
+    enterpriseSearchUrl = undefined;
+  });
+
+  const render = () => {
+    return mount(
+      <KibanaContext.Provider
+        value={{
+          http: {},
+          enterpriseSearchUrl,
+          setBreadcrumbs: jest.fn(),
+        }}
+      >
+        <EngineOverviewHeader />
+      </KibanaContext.Provider>
+    );
+  };
```

Comment:
I'd like to pull this out to a reusable helper function if possible. Something like this:

```jsx
global.mountWithContext = (component, context) => {
  const defaultContext = {
    http: jest.fn(),
    setBreadcrumbs: jest.fn(),
    enterpriseSearchUrl: 'http://localhost:3002',
  };

  return mount(
    <KibanaContext.Provider value={{ ...defaultContext, ...context }}>
      {component}
    </KibanaContext.Provider>
  );
};
```

I'd strongly prefer `mountWithContext()` over `render()` - `render()` is already an existing OOTB function within Enzyme and one that most people find confusing vs shallow/mount and that I personally prefer to avoid.

My thought is that this could get exported from `index.test.ts`, but maybe `helpers.test.js` is better? 🤷 I'd normally put in a Jest [setupFile](https://jestjs.io/docs/en/configuration.html#setupfiles-array) but unfortunately since we're inheriting Jest settings from Kibana, I don't think we can make this a global helper without an import.

Reply:
I like it.

Reply:
I created a `test_utils/helpers.tsx` file, which seemed consistent with what Kibana does with other test helpers like their enzyme helper. a7b0acd579ac21f46e165ecc49ec1a0643b10d1f